### DISCUSSION TITLE
about gomavlib and mavsdk-go

### DISCUSSION THREAD

**Discussion by itstmyi** [bluenviron/gomavlib#29] [Q&A]
May I ask a stupid question ?

What's the difference between gomavlib and mavsdk-go( github.com/mavlink/MAVSDK-Go ) ?

I am learning something on PX4( flight control software) and don't know which one  should I choose .
Thanks for any information .

**Comment by aler9** [comment]
Hello,

MAVSDK-Go interacts with the the MAVSDK daemon, that must be running in the background, and talks to it through gRPC.

gomavlib is a native library, it doesn't require anything and talks directly to any Mavlink node through Mavlink.

**Comment by itstmyi** [comment]
Get it . Thank you very much .
